sched/scheduler/stateful_scheduler.go

```diff
@@ -238,11 +243,17 @@ func (s *statefulScheduler) scheduleTasks() {
 		s.asyncRunner.RunAsync(
 			func() error {
 				log.Println("Starting task", taskId, " command:", strings.Join(taskDef.Argv, " "))
-				return runTaskAndLog(
+				status, err := runTaskAndLog(
 					saga,
 					wf,
 					taskId,
 					taskDef)
+				if err == nil && status.State == runner.COMPLETE {
+					s.stat.Counter("schedCompletedTaskCounter").Inc(1)
```

Comment:
For code simplicity/maintainability we should log this closer to the code where the request is being made to the worker/runner.

Why not just log these in runTaskAndLog instead of propagating this information all the way back?